src/cursor/cur_file.c

```diff
@@ -190,6 +202,7 @@ __curfile_search(WT_CURSOR *cursor)
     cbt = (WT_CURSOR_BTREE *)cursor;
     CURSOR_API_CALL(cursor, session, search, cbt->btree);
     WT_ERR(__cursor_checkkey(cursor));
+    WT_ERR(__cursor_copy_release_value(cursor));
```

Comment:
I'm probably missing something, but can't the key be set here? A previous call that returned a key, followed by a search, wouldn't we get here and need to call `__cursor_copy_release_key()`?

If I'm correct, that change would apply to other places, of course.

Reply:
You are correct.  I was thinking about some simple use cases and missed this.  I have fixed this.

Reply:
I did that, adding a pair to all the methods.  Since they are all pairs, I refactored, made them a single method.  In doing this, I found an error in the release method:  there's one case where the original `WT_ITEM` was being cleared (by a call to `__wt_buf_free`).  I went with the cleanest way to fix it was (I think), but it's potentially more wasteful.  If the item points to something external, it always gets a local copy made.  I don't think we're worried about waste here, but I'd appreciate it if you took another look at `__wt_cursor_copy_release_item`